presidio-anonymizer/presidio_anonymizer/anonymizer_engine.py

```diff
@@ -75,3 +81,28 @@ def __validate_position_over_text(self, analyzer_result, text_len):
                 f"Invalid analyzer result: '{analyzer_result}', "
                 f"original text length is only {text_len}."
             )
+
+    def __extract_anonymizer(self, transformation):
+        anonymizer = transformation.get("anonymizer")()
+        # if the anonymizer is not valid, a InvalidParamException
+        anonymizer.validate(params=transformation)
+        return anonymizer
+
+    def __anonymize_section(
```

Comment:
I agree it should have its own method but it is really overloaded in the parameters.

Reply:
agree, the purpuse is to make the main loop more readable